pkg/policy/mapstate.go

```diff
@@ -933,12 +933,14 @@ func (ms *mapState) denyPreferredInsertWithChanges(newKey Key, newEntry MapState
 					// must be added.
 					denyKeyCpy := k
 					denyKeyCpy.Identity = newKey.Identity
-					l3l4DenyEntry := NewMapStateEntry(k, v.DerivedFromRules, false, true, DefaultAuthType, AuthTypeDisabled)
+					// Note that we let the new key own this copy, since only
+					// the new key can be incrementally deleted.
+					l3l4DenyEntry := NewMapStateEntry(newKey, v.DerivedFromRules, false, true, DefaultAuthType, AuthTypeDisabled)
```

Comment:
I think there are other instances above where the owner of the synthetic entry is a deny key or a key with `0` identity. Should probably flip the ownership of those as well in the same way.

Reply:
Basically go through each line that adds a synthetic entry and ask if the first argument of `NewMapStateEntry` either:
- has a `0` identity, or
- is a deny key

If yes flip the ownership.

Reply:
I tried to apply this, but it seemed to only apply to one other case.

If both k and newKey was deny/had zero identity I think it doesn't actually matter which owns the synthesized entry, so I left those.